lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -341,24 +336,19 @@ instance Buildable (PartialTx era) where
         cardanoTxF tx' = pretty $ pShow tx'
 
 -- | Assumptions about the UTxO which are needed for coin-selection.
---
--- In practice turns out to be
---
--- @
---  data UTxOAssumptions
---      = AllKeyPaymentCredentials
---      | AllByronKeyPaymentCredentials
---      | AllScriptPaymentCredentialsFrom ScriptTemplate scriptLookup
--- @
---
--- however representing it as such is inconvenient at the moment.
-data UTxOAssumptions = forall k ktype. UTxOAssumptions
-    { txLayer :: TransactionLayer k ktype SealedTx
-    -- TODO: Replace with smaller and smaller parts of 'TransactionLayer'
-    , inputScriptLookup :: Maybe (W.Address -> CA.Script KeyHash)
-    , inputScriptTemplate :: Maybe ScriptTemplate
-    , txWitnessTag :: TxWitnessTag
-    }
+data UTxOAssumptions
+    = AllKeyPaymentCredentials
+    -- ^ Assumes all 'UTxO' entries have addresses with the post-Shelley
+    -- key payment credentials.
```

Comment:
I would suggest using the term "Shelley" here, as:
- the convention of "Byron" versus "Shelley" is already established in the ledger types.
- the other constructor specifies `Byron`.

Additionally, if this constructor is compatible with key payment credentials of the Shelley era, then perhaps we can change the wording to "Shelley era or later" rather than "post-Shelley", as "post-X" can imply that "X" is not included.

```suggestion
    = AllShelleyKeyPaymentCredentials
```suggestion
    -- ^ Assumes all 'UTxO' entries have addresses with key payment
    -- credentials corresponding to the Shelley era or later.
``````